atlasdb-buggify/src/main/java/com/palantir/atlasdb/buggify/impl/DefaultBuggifyFactory.java

```diff
@@ -19,13 +19,33 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.palantir.atlasdb.buggify.api.Buggify;
 import com.palantir.atlasdb.buggify.api.BuggifyFactory;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
 import java.security.SecureRandom;
 import java.util.function.DoubleSupplier;
 
 public final class DefaultBuggifyFactory implements BuggifyFactory {
-    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
+    private static final SafeLogger log = SafeLoggerFactory.get(DefaultBuggifyFactory.class);
+    private static final SecureRandom SECURE_RANDOM;
     public static final BuggifyFactory INSTANCE = new DefaultBuggifyFactory();
 
+    static {
+        SecureRandom randomInstance;
+        try {
+            // Antithesis's implementation of /dev/urandom is deterministic and something controlled by their fuzzer,
+            // which allows them to explore the state space. We thus choose the NativePRNG algorithm which is backed by
+            // /dev/urandom. This string is a reserved identifier for the algorithm, following
+            // https://docs.oracle.com/en/java/javase/17/docs/specs/security/standard-names.html#securerandom-number-generation-algorithms
+            randomInstance = SecureRandom.getInstance("NativePRNG");
```

Comment:
iirc `NativePRNG` uses a combination of both and is likely already the default, but can be pointed at exclusively `/dev/urandom` by configuring the jvm with `-Djava.security.egd=file:/dev/./urandom`. Sorry if I've misremembered, though!

Reply:
Ah my bad. Yeah both devices are actually deterministic in their environment - the main thing was to make sure the algorithm selected is one that actually accesses entropy on generating numbers as opposed to only on seeds.